sources/platform/actors/development/builds_and_runs/state_persistence.md

```diff
@@ -1,51 +1,70 @@
 
-A migration is when a process running on a server has to stop and move to another. All in-progress processes on the current server are stopped. Unless you have saved your state, the Actor run will restart on the new server. For example, if a request in your [request queue](../../../storage/request_queue.md) has not been updated as **crawled** before the migration, it will be crawled again.
+## Undersanding migrations
 
-**When a migration event occurs, you only have a few seconds to save your work.**
+A migration occurs when a process running on one srever must stop and move to another. During this process:
 
-## Why do migrations happen
+- All in-progress processes on the current server are stopped
+- Unless you've saved your state, the Actor run will restart on the new server
+- You only have a few seconds to save your work when a migration event occurs
 
-- To optimize server workloads.
-- When a server crashes (unlikely).
-- When we release new features and fix bugs.
+### Causes of migration
 
-## How often do migrations occur
+Migrations can happen for several reasons:
 
-Migrations have no specific interval at which they happen. They are caused by the [above events](#why-do-migrations-happen), so they can happen at any time.
+- Server workload optimization
+- Server crashes (rare)
+- New feature releases and bug fixes
 
-## Why is state lost during migration
+### Frequency of migrations
 
-Unless instructed to save its output or state to a [storage](../../../storage/index.md), an Actor keeps them in the server's memory. When it switches servers, the run loses access to the previous server's memory. Even if data were saved on the server's disk, we would also lose access to that.
+Migrations don't follow a specific schedule. They can occur at any time due to the events mentioned above.
 
-## How to persist state
+## Why state is lost during migration
+
+By default, an Actor keeps its output and state in the server's memory. During a server switch, the run loses access to the previous server's memory. Even if data were saved on the server's disk, access to that would also be lost.
+
+## Implementing state persistence
+
+The [Apify SDKs](/sdk) handle state persistence automatically.
+
+In JavaScript, this is done using the `migrating` and `persistState` events in the [PlatformEventManager](/sdk/js/api/apify/class/PlatformEventManager).
+
+- The `persistState` event prompts SDK components to save their state at regular intervals
+- The `migrating` event is triggered just before a migration occurs.
+
+In Python, state persistence is handled using the `Actor.on()` method and the migrating event, similar to JavaScript. The Apify SDK for Python provides mechanisms to save and retrieve state data.
+
+- The `migrating` event is triggered just before a migration occurs, allowing you to save your state.
+- To retrieve previously saved state, you can use the `Actor.get_value()` method.
```

Comment:
I believe you could link the `PlatformEventManager` API reference of both JS and Python implementations.

You can also mention, that the event manager is encapsulated with Actor class, and when you want to interact with that you should use `Actor.on` and `Actor.off`. And it will use the underlying event manager instance in Actor.

Or you do not have to mention `EventManagers` at all and go with just the shortcuts.